proto/regen/ecocredit/v1alpha1/query.proto

```diff
@@ -177,3 +195,71 @@ message QueryCreditTypesResponse {
   // list of credit types
   repeated CreditType credit_types = 1;
 }
+
+message QueryBasketRequest {
+  // basket_denom represents the denom of the basket to query. 
+  string basket_denom = 1;
+}
+
+message QueryBasketResponse {
+  // basket is the queried basket.
+  Basket basket = 1;
+}
+
+message QueryBasketsRequest {
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+
+message QueryBasketsResponse {
+  // baskets are the fetched baskets.
+  repeated Basket baskets = 1;
+
+  // pagination defines the pagination in the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+message QueryBasketCreditsRequest {
+  // basket_denom is the basket denom to query credits for.
+  string basket_denom = 1;
+
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+message QueryBasketCreditsResponse {
+  // credits are the credits inside the basket.
+  repeated BasketCredit credits = 1;
+
+  // pagination defines the pagination in the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// Basket defines a credit basket.
+message Basket {
+  // curator is the address of the basket curator who is able to change certain
+  // basket settings.
+  string curator = 1;
+
+  // name will be used to create a bank denom for this basket token of the form
+  // ecocredit:{curator}:{name}.
+  string name = 2;
+
+  // display_name will be used to create a bank Metadata display name for this
+  // basket token of the form ecocredit:{curator}:{display_name}.
+  string display_name = 3;
+
+  // exponent is the exponent that will be used for denom metadata. An exponent
+  // of 6 will mean that 10^6 units of a basket token should be displayed
+  // as one unit in user interfaces.
+  uint32 exponent = 4;
+
+  // basket_criteria is the criteria by which credits can be added to the
+  // basket. Basket criteria will be applied in order and the first criteria
+  // which applies to a credit will determine its multiplier in the basket.
+  repeated BasketCriteria basket_criteria = 5;
+
+  bool retire_on_take = 6;
```

Comment:
This might need to be updated to `disable_auto_retire`:

https://github.com/regen-network/regen-ledger/pull/610/files#r746245166